check-approved-requests.py

```diff
@@ -0,0 +1,229 @@
+
+Each run of the script will:
+    - Check for rows from the spreadsheet which have been approved, but the
+      helpdesk hasn't been notified
+    - Generate an email to the helpdesk in the correct format
+    - Update the spreadsheet to indicate the helpdesk was notified
+    - (Optional) log the processed requests to the specified log file
+"""
+import argparse
+import ConfigParser
+from datetime import datetime
+from spreadsheet import Spreadsheet
+from message import TemplateMessage
+from config import set_config_file
+
+
+def parse_user_row(cells):
+    """Parse the new approved User/Project request row"""
+    email = cells[3].replace(u'\xa0', ' ').strip()
+    user_info = {'user_name': email,
+                 'user_email': email,
+                 'first_name': cells[4],
+                 'last_name': cells[5]}
+    
+    comment = 'User requested the following access:'.format(**user_info)
+    
+    REQUEST = "\n - {req}: `{detail}`"
+
+    if cells[6] == 'No':
+        req_type = 'new OpenStack user account'
+        comment += REQUEST.format(req=req_type, detail=email)
+    
+    if cells[14] != '':
+        req_type = 'new Openstack project'
+        comment += REQUEST.format(req=req_type, detail=cells[14])
+
+        try:
+            user_list = cells[16]
+            req_type = "add existing users to new project"
+            comment += REQUEST.format(req=req_type, detail=user_list)
+        except IndexError:
+            # it's OK for cells[16] not to exist at all
+            pass
+        
+        # project description at the end because it might be long
+        comment += "\n\nNew Project Description:\n{}".format(cells[15])
+    
+    elif cells[17] != '':
+        req_type = 'access to existing OpenStack project'
+        comment += REQUEST.format(req=req_type, detail=cells[17])
+        
+    user_info['comment'] = comment
+    return user_info
+
+
+def parse_quota_row(cells):
+    """Parse the new approved Quota request row"""
+    # NOTE: cells[16] is a required field in the Google Form, so it is safe
+    # to assume cells[0:15] exists.
+    email = cells[3].replace(u'\xa0', ' ').strip()
+    user_info = {'user_name': email,
+                 'user_email': email,
+                 'first_name': cells[4],
+                 'last_name': cells[5]}
+
+    comment = 'User requested new quotas for project: {}'.format(cells[7])
+    comment += "\n\nRequest Details:\n"
+    
+    # FIXME: this code block is lifted straight from set-quotas.py,
+    # farm it out to a function somewhere to streamline updates.  Possibly
+    # also update to use a list and for i in range().
+    quotas = {'instances': cells[10],
+              'cores': cells[11],
+              'ram': cells[12],
+              'floatingip': cells[13],
+              'volumes': cells[14],
+              'snapshots': cells[15],
+              'gigabytes': cells[16]}
+    unchanged_quotas = [q for q in quotas if quotas[q] == '']
```

Comment:
That's true and it would violate a certain pep8 rule since it is too long I guess.